src/config.rs

```diff
@@ -295,6 +298,33 @@ impl Context {
                 let value = value.map(improve_single_line_input);
                 self.sql.set_raw_config(key, value.as_deref()).await?;
             }
+            Config::SentboxWatch => {
+                self.sql.set_raw_config(key, value).await?;
+                if config_to_bool(value) {
+                    self.sql
+                        .set_raw_config(Config::OnlyFetchMvbox, Some("0"))
+                        .await?;
+                }
+            }
+            Config::MvboxMove => {
+                self.sql.set_raw_config(key, value).await?;
+                if !config_to_bool(value) {
+                    self.sql
+                        .set_raw_config(Config::OnlyFetchMvbox, Some("0"))
+                        .await?;
+                }
+            }
+            Config::OnlyFetchMvbox => {
+                self.sql.set_raw_config(key, value).await?;
+                if config_to_bool(value) {
+                    self.sql
+                        .set_raw_config(Config::SentboxWatch, Some("0"))
+                        .await?;
+                    self.sql
+                        .set_raw_config(Config::MvboxMove, Some("1"))
+                        .await?;
+                }
```

Comment:
how does all this logic interact with UIs?  Currently there are a bunch of sliders, now if someone sets one of these the UI is supposed to re-fetch all the settings shown in order to reset all the sliders in the right positions?  Does this seem like a pitfall?

Reply:
> Currently there are a bunch of sliders, now if someone sets one of these the UI is supposed to re-fetch all the settings shown in order to reset all the sliders in the right positions?

Yes. Should this be made clearer in `deltachat.h`?

>  Does this seem like a pitfall?

Kind of, but do you have a better idea?

Reply:
Would calling the new option `FetchEnabledOnly` avoid having multiple boolean options that need to interact?  On first glance I think it might?

Reply:
So, you mean, making an option that disables the `scan_folders` feature and only fetches from folders that are watched?
The thing is, we still need to check the Spam folder from time to time (also, currently - although we could change this - the Inbox folder is still watched/idled, the messages there are just ignored).

Also, the user might not know that in order for the mvbox to be watched, they need to enable mvbox_move - they might disable this setting because they think "there is no reason for DC to move the messages to the mvbox, I'm doing this with my server rule". And then no folders would be looked at at all.

Reply:
Refetching all the options requires changes in all the UIs. It's also not good that toggling this option on and off does not result in the same state as before.

I would prefer changing the setting not to change other options, but have methods like `should_watch_inbox` that calculates `mvbox_move or only_fetch_mvbox` and a method `should_watch_sentbox` calculating `sentbox_watch and not only_fetch_mvbox` that are used internally. So if you enable the new option, old options are ignored but their state is not changed.

Reply:
So @Hocuri has been giving me some more info and the new options look like:
```
- Watch Sent Folder
- Send Copy to Self (not relevant here)
- Automatic Moves to DeltaChat folder
```

For the record the old options were:
```
- watch inbox folder
- watch sent folder
- watch DeltaChat folder
- send copy to self
- automatic moves to deltachat folder
```
and the "watch dc folder" and "move to dc folder" have been merged.

So I don't think we need all the interdependencies, in particular the new option should not change `SendboxWatch` or `MvboxMove`.  In fact i think someone who has server-side rules would not be enabling `MvboxMove`.

All the new option needs to do is skip watching the inbox.  So maybe call it `DontWatchInbox` "Do not watch inbox" and it works again as a simple boolean?

Reply:
We still want to watch the movebox to receive messages immediately.

Reply:
yes, a `DontWatchInbox` setting allows that though does it not?  Or is this more a general observation rather than a problem you see with my proposal?

Reply:
> All the new option needs to do is skip watching the inbox. So maybe call it DontWatchInbox "Do not watch inbox" and it works again as a simple boolean?

No, it also needs to skip fetching messages from unwatched folders in `scan_folders()` (except for the Spam folder, which should always be fetched from time to time). And we would somehow need to communicate to the user that they need to enable MvboxMove.

> In fact i think someone who has server-side rules would not be enabling MvboxMove.

Correct, and that's the problem - not enabling MvboxMove means not watching the mvbox because the options were merged.

> I would prefer changing the setting not to change other options, but have methods like `should_watch_inbox` that calculates `mvbox_move or only_fetch_mvbox` and a method `should_watch_sentbox` calculating `sentbox_watch and not only_fetch_mvbox` that are used internally. So if you enable the new option, old options are ignored but their state is not changed.

This has the risk that someone might accidentally use `get_config(SentboxWatch)` instead of `should_watch_sentbox()` because they don't know about this and introduce a bug.

It also has the slight performance disadvantage of two database accesses instead of one.

All in all, I'm like -0.2 or something on this idea compared to the current implementation.

Reply:
I still believe we really need to make sure these options don't interact awkwardly.  `WatchMvboxNotInbox` and "Watch DeltaChat folder instead of Inbox".  If unset we would still fetch the DeltaChat folder but that's fine I think.

It feels like the desired behaviour is simple it is just really awkward to express nicely both as an option and to the user.

Reply:
> `WatchMvboxNotInbox` and "Watch DeltaChat folder instead of Inbox". If unset we would still fetch the DeltaChat folder but that's fine I think.

This doesn't express that this config also disables scanning all folders (except for the Spam folder).

> It feels like the desired behaviour is simple it is just really awkward to express nicely both as an option and to the user.

+1 :100:

> I still believe we really need to make sure these options don't interact awkwardly.

FWIW, when using the Android PR, to me personally it felt natural that when setting only_fetch_mvbox, mvbox_move and sentbox_watch are possibly changed and then greyed out. So, from a UI perspective this seems OK, esp. as only rather advanced users will use it.

I agree that from a code/architecture standpoint, it's not nice that changing this config automatically changes other configs; also it's not nice that the UI has to refresh these configs from the DB; but both seems OK-ish to me. Obviously, if y'all want, I can implement @link2xt's idea (the one I was like -0.2 on).

Reply:
I don't like that UIs have to refresh all options because it definitely breaks KDeltaChat and likely breaks Desktop and the typical React workflow, because normally you have an event telling you when things update and desktop will need one more internal fake event for this.